custom_components/london_tfl/config_flow.py

```diff
@@ -134,10 +136,32 @@ async def async_step_station(self, user_input: Optional[Dict[str, Any]] = None):
                             "JSON.", exc_info=1)
             errors["base"] = "request"
 
+        description_placeholders = {"extra_description": ""}
+        extra_fields = {}
+
+        if (
+            self.data["lastMethod"] == "national-rail"
+            and self.data["lastLine"] != "thameslink"
+        ):
+            # if a user already supplied a token in this specific flow, retrieve it
+            kwargs = {}
+            for stop in self.data[CONF_STOPS]:
+                token = stop.get(CONF_NR_API_KEY)
+                if token is not None:
+                    kwargs["default"] = token
+
+            extra_fields[vol.Required(CONF_NR_API_KEY, **kwargs)] = cv.string
+            description_placeholders["extra_description"] = """
+This line is not supported by the TfL API and requires an extra API token in order to see departure times.

```

Comment:
Not a fan of this being hardcoded and not in translations. We can leave it as is for now, but if you know a quick way to get it into en.json, that would be nice.

Reply:
I looked into the docs for a while but couldn't find a way to do it properly. The only way I can think of is to duplicate the step, so there is one specifically for LDBWS and the previous one.